src/main/java/com/oneops/proxy/web/ErrorController.java

```diff
@@ -52,8 +53,27 @@ public ErrorController(ErrorAttributes errAttrs, ServerProperties serverProperti
   public @ResponseBody ResponseEntity<ErrorResponse> errorResponse(
       HttpServletRequest req, HttpServletResponse res) {
     Map<String, Object> body = getErrorAttributes(req, isIncludeStackTrace(req, MediaType.ALL));
-    HttpStatus status = getStatus(req);
 
+    /*---------------Update status code and format error respose for RequestRejectedException------------------*/
+    Object message = null;
+    Object path = null;
+    String exceptionName = body.get("exception").toString();
+    final String REQUEST_REJECTED_EXCEPTION = "RequestRejectedException";
+
+    if (exceptionName.contains(REQUEST_REJECTED_EXCEPTION)) {
+      try{
+        message = URLDecoder.decode(body.get("message").toString(), "UTF-8");
+        path = URLDecoder.decode(body.get("path").toString(), "UTF-8");
+      }catch(UnsupportedEncodingException ue){}
+      message = message.toString().replace(exceptionName +":", "");
+      body.put("message", message);
+      body.put("path",path);
+      body.put("status", HttpStatus.BAD_REQUEST.value());
+      body.put("error",HttpStatus.BAD_REQUEST.getReasonPhrase());
+      req.setAttribute("javax.servlet.error.status_code", HttpStatus.BAD_REQUEST.value());
```

Comment:
Do we need to rewrite the HttpRequest also? All we want is the error response to populate with proper values. Make minimal changes to return the proper response.

Reply:
Had discussed on this. updated code based on it.